cmd/limactl/disk.go

```diff
@@ -0,0 +1,305 @@
+	}
+
+	return nil
+}
+
+func newDiskListCommand() *cobra.Command {
+	var diskListCommand = &cobra.Command{
+		Use: "list",
+		Example: `
+To list existing disks:
+$ limactl disk list
+`,
+		Short:   "List existing Lima disks",
+		Aliases: []string{"ls"},
+		Args:    cobra.NoArgs,
+		RunE:    diskListAction,
+	}
+	diskListCommand.Flags().Bool("json", false, "JSONify output")
+	return diskListCommand
+}
+
+func diskListAction(cmd *cobra.Command, args []string) error {
+	jsonFormat, err := cmd.Flags().GetBool("json")
+	if err != nil {
+		return err
+	}
+
+	allDisks, err := store.Disks()
+	if err != nil {
+		return err
+	}
+
+	if jsonFormat {
+		for _, diskName := range allDisks {
+			disk, err := store.InspectDisk(diskName)
+			if err != nil {
+				logrus.WithError(err).Errorf("disk %q does not exist?", diskName)
+				continue
+			}
+			j, err := json.Marshal(disk)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintln(cmd.OutOrStdout(), string(j))
+		}
+		return nil
+	}
+
+	w := tabwriter.NewWriter(cmd.OutOrStdout(), 4, 8, 4, ' ', 0)
+	fmt.Fprintln(w, "NAME\tSIZE\tDIR\tIN USE BY")
+
+	if len(allDisks) == 0 {
+		logrus.Warn("No disk found. Run `limactl disk create DISK --size SIZE` to create a disk.")
+	}
+
+	for _, diskName := range allDisks {
+		disk, err := store.InspectDisk(diskName)
+		if err != nil {
+			logrus.WithError(err).Errorf("disk %q does not exist?", diskName)
+			continue
+		}
+		fmt.Fprintf(w, "%s\t%s\t%s\t%s\n", disk.Name, units.BytesSize(float64(disk.Size)), disk.Dir, disk.Instance)
+	}
+
+	return w.Flush()
+}
+
+func newDiskDeleteCommand() *cobra.Command {
+	var diskDeleteCommand = &cobra.Command{
+		Use: "delete DISK [DISK, ...]",
+		Example: `
+To delete a disk:
+$ limactl disk delete DISK
+
+To delete multiple disks:
+$ limactl disk delete DISK1 DISK2 ...
+`,
+		Aliases: []string{"remove", "rm"},
+		Short:   "Delete one or more Lima disks",
+		Args:    cobra.MinimumNArgs(1),
+		RunE:    diskDeleteAction,
+	}
+	diskDeleteCommand.Flags().Bool("force", false, "force delete")
+	return diskDeleteCommand
+}
+
+func diskDeleteAction(cmd *cobra.Command, args []string) error {
+	force, err := cmd.Flags().GetBool("force")
+	if err != nil {
+		return err
+	}
+
+	for _, diskName := range args {
+		if force {
+			disk, err := store.InspectDisk(diskName)
+			if err != nil {
+				if errors.Is(err, fs.ErrNotExist) {
+					logrus.Warnf("Ignoring non-existent disk %q", diskName)
+					continue
+				}
+				return err
+			}
+
+			if err := deleteDisk(disk); err != nil {
+				return fmt.Errorf("failed to delete disk %q: %w", diskName, err)
+			}
+			logrus.Infof("Deleted %q (%q)", diskName, disk.Dir)
+			continue
+		}
+
+		disk, err := store.InspectDisk(diskName)
+		if err != nil {
+			if errors.Is(err, fs.ErrNotExist) {
+				logrus.Warnf("Ignoring non-existent disk %q", diskName)
+				continue
+			}
+			return err
+		}
+		if disk.Instance != "" {
+			return fmt.Errorf("cannot delete disk %q in use by instance %q", disk.Name, disk.Instance)
```

Comment:
It feels inconsistent to me that a locked disk will abort the command, but a non-existing disk is simply skipped:

```
limactl disk rm foo bar
```

If `foo` doesn't exist, `bar` will still be deleted. But if `foo` exists and is locked, then `bar` will not be deleted (and there is no message to the effect that additional disks have been skipped due to the error).

I think this the locked disk should be logged as an error, but additional disks should still be processed.

And then I wonder if failures if `deleteDisk` should be treated the same way too: log an error, but continue to attempt deleting the remaining disks?

Of course the command must exit with a non-zero status code if any disk couldn't be deleted. I'm not sure how non-existing disks should be treated: should they also cause a non-zero error?

@AkihiroSuda WDYT?

Reply:
Can you open an issue for this? I think we should keep this behavior consistent between `limactl disk delete` and `limactl delete`, and I think this needs some more discussion.